cluster/src/main/java/org/apache/iotdb/cluster/entity/raft/DataStateMachine.java

```diff
@@ -82,59 +77,71 @@ public DataStateMachine(String groupId, PeerId peerId) {
-
-      LOGGER.debug(String.format("State machine batch size(): %d", planBytes.size()));
-
-      /** handle batch plans(planBytes.size() > 0) or single plan(planBytes.size()==1) **/
-      for (byte[] planByte : planBytes) {
-        try {
-          PhysicalPlan plan = PhysicalPlanLogTransfer.logToOperator(planByte);
-
-          LOGGER.debug(String.format("OperatorType :%s", plan.getOperatorType()));
-          /** If the request is to set path and sg of the path doesn't exist, it needs to run null-read in meta group to avoid out of data sync **/
-          if (plan.getOperatorType() == OperatorType.CREATE_TIMESERIES && !checkPathExistence(
-              ((MetadataPlan) plan).getPath().getFullPath())) {
-            RaftUtils.handleNullReadToMetaGroup(status);
-            if(!status.isOk()){
-              continue;
-            }
-          }
-          qpExecutor.processNonQuery(plan);
-          if (closure != null) {
-            response.addResult(true);
-          }
-        } catch (ProcessorException | IOException | PathErrorException e) {
-          LOGGER.error("Execute physical plan error", e);
-          status = new Status(-1, e.getMessage());
-          if (closure != null) {
-            response.addResult(false);
+      iterator.next();
+    }
+  }
+
+  /**
+   * Apply a single raft task. If this node is the leader of state machine's data group, apply the
+   * task in thread.
+   *
+   * @param closure if this node is leader, closure is not null.
+   * @param data Request data
+   */
+  private void applySingleTask(Closure closure, ByteBuffer data) {
+    BasicResponse response = (closure == null) ? null : ((ResponseClosure) closure).getResponse();
+    DataGroupNonQueryRequest request = null;
+    try {
+      request = SerializerManager.getSerializer(SerializerManager.Hessian2)
+          .deserialize(data.array(), DataGroupNonQueryRequest.class.getName());
+    } catch (final CodecException e) {
+      LOGGER.error("Fail to decode IncrementAndGetRequest", e);
+    }
+
+    Status status = Status.OK();
+    assert request != null;
+
+    List<byte[]> planBytes = request.getPhysicalPlanBytes();
+
+    LOGGER.debug(String.format("State machine batch size(): %d", planBytes.size()));
+
+    /** handle batch plans(planBytes.size() > 0) or single plan(planBytes.size()==1) **/
+    for (byte[] planByte : planBytes) {
+      try {
+        PhysicalPlan plan = PhysicalPlanLogTransfer.logToOperator(planByte);
+
+        LOGGER.debug(String.format("OperatorType :%s", plan.getOperatorType()));
+        /** If the request is to set path and sg of the path doesn't exist, it needs to run null-read in meta group to avoid out of data sync **/
+        if (plan.getOperatorType() == OperatorType.CREATE_TIMESERIES && !checkPathExistence(
+            ((MetadataPlan) plan).getPath().getFullPath())) {
+          RaftUtils.handleNullReadToMetaGroup(status);
+          if(!status.isOk()){
+            continue;
```

Comment:
`continue` means this plan can not be executed, right ?

Reply:
yes

Reply:
Then, if one of the plan cannot be executed, how do users know that ？

Reply:
It needs to add a false result to response.fixed